auth/credentials/api.go

```diff
@@ -0,0 +1,62 @@
+package credentials
```

Comment:
nit: credentials and auth seems similar terms, what about using serviceaccount or similar  as package name? in that way func could be serviceaccount.NewClientWithTokenSource, NewHTTPClient, etc.

Reply:
Pinged you offline on the documentation to discuss potential renames. We can apply any name we would decide upon post merge.